File: src/lib/api/tracks.ts
import { invoke } from "@tauri-apps/api/core";

export async function setTrackVolume(trackIndex: number, volume: number) {
  return invoke("set_track_volume", { trackIndex, volume });
}

export async function setTrackMute(trackIndex: number, mute: boolean) {
  return invoke("set_track_mute", { trackIndex, mute });
}

export async function setTrackSolo(trackIndex: number, solo: boolean) {
  return invoke("set_track_solo", { trackIndex, solo });
}

export async function setTrackPan(trackIndex: number, pan: number) {
  return invoke("set_track_pan", { trackIndex, pan });
}

File: src-tauri/src/commands/tracks.rs
use std::sync::Arc;
use tauri::State;

use crate::engine::EngineAdapter;

#[tauri::command]
pub fn set_track_volume(
    engine: State<'_, Arc<dyn EngineAdapter>>,
    track_index: i32,
    volume: f32,
) -> Result<(), String> {
    engine.set_track_volume(track_index, volume)
}

#[tauri::command]
pub fn set_track_mute(
    engine: State<'_, Arc<dyn EngineAdapter>>,
    track_index: i32,
    mute: bool,
) -> Result<(), String> {
    engine.set_track_mute(track_index, mute)
}

#[tauri::command]
pub fn set_track_solo(
    engine: State<'_, Arc<dyn EngineAdapter>>,
    track_index: i32,
    solo: bool,
) -> Result<(), String> {
    engine.set_track_solo(track_index, solo)
}

#[tauri::command]
pub fn set_track_pan(
    engine: State<'_, Arc<dyn EngineAdapter>>,
    track_index: i32,
    pan: f32,
) -> Result<(), String> {
    engine.set_track_pan(track_index, pan)
}
